packages/stripe/src/Adaptor.js

```diff
@@ -0,0 +1,313 @@
+import { expandReferences } from '@openfn/language-common/util';
+import * as util from './Utils';
+
+/**
+ * State object
+ * @typedef {Object} HttpState
+ * @property data - the parsed response body
+ * @property response - the response from the HTTP server, including headers, statusCode, body, etc
+ * @property references - an array of all previous data objects used in the Job
+ **/
+
+/**
+ * Payment Intents Options object. See {@link https://docs.stripe.com/api/payment_intents/list|Stripe Payment Intents API documentation} for more details.
+ * @typedef {Object} PaymentIntentsOptions
+ * @public
+ * @property {object} created - An object of created fields for filtering the data.
+ * @property {string} ending_before - Object string ID that defines your place in the list. Example: `obj_bar` in order to fetch the previous page of the list.
+ * @property {string} starting_after - Object string ID that defines your place in the list. Example: `obj_foo` in order to fetch the next page of the list.
+ * @property {number} limit - A limit on the number of objects to be returned. Limit can range between 1 and 100, and the default is 10.
+ */
+
+/**
+ * Get payments from Stripe
+ * @example <caption>List all payment intents</caption>
+ * listPaymentIntents();
+ * @example <caption>List payment intents with options</caption>
+ * listPaymentIntents({limit:3});
+ * @function
+ * @public
+ * @param {PaymentIntentsOptions} options - Optional payment intent request options
+ * @returns {Operation}
+ * @state {HttpState}
+ */
+export function listPaymentIntents(options = {}) {
+  return async state => {
+    const [resolvedoptions] = expandReferences(state, options);
+
+    const response = await util.request(
+      state.configuration,
+      'GET',
+      '/payment_intents',
+      {
+        query: {
+          ...resolvedoptions,
+        },
+      }
+    );
+
+    return util.prepareNextState(state, response);
+  };
+}
+
+/**
+ * Get a Stripe payment intent
+ * @example
+ * getPaymentIntent('pi_3RxS5EEAUr6ipfDB0aKo3moC');
+ * @function
+ * @public
+ * @param {string} paymentIntentId - The ID of the payment intent to retrieve.
+ * @returns {Operation}
+ * @state {HttpState}
+ */
+export function getPaymentIntent(paymentIntentId) {
+  return async state => {
+    const [resolvedPaymentIntentId] = expandReferences(state, paymentIntentId);
+
+    const response = await util.request(
+      state.configuration,
+      'GET',
+      `/payment_intents/${resolvedPaymentIntentId}`,
+      {}
+    );
+
+    return util.prepareNextState(state, response);
+  };
+}
+
+/**
+ * Customers Options object. See {@link https://docs.stripe.com/api/customers/list|Stripe Customers API documentation} for more details.
+ * @typedef {Object} CustomersOptions
```

Comment:
We shouldn't duplicate Stripe documentation: it should be OK just to link to Stripe itself. If we have own own options, they MUST be documented. If our users are particularly reliant on certain options, then we can include those as examples. But generally we shouldn't be having so much documentation - if Stripe makes a change, then our docs will become out of date.